stock_partner_delivery_window/models/res_partner.py

```diff
@@ -49,19 +57,13 @@ def get_delivery_windows(self, day_name=None):
         :param day: The day name (see time.weekday, ex: 0,1,2,...)
         :return: dict partner_id: delivery_window recordset
         """
-        res = {}
+        self.ensure_one()
         domain = [("partner_id", "in", self.ids)]
         if day_name is not None:
             week_day_id = self.env["time.weekday"]._get_id_by_name(day_name)
             domain.append(("time_window_weekday_ids", "in", week_day_id))
         windows = self.env["partner.delivery.time.window"].search(domain)
-        for window in windows:
-            if not res.get(window.partner_id.id):
-                res[window.partner_id.id] = self.env[
-                    "partner.delivery.time.window"
-                ].browse()
-            res[window.partner_id.id] |= window
-        return res
+        return windows
```

Comment:
Why did you changed this? The docstring says _"Return the list of delivery windows by partner id for the given day"_, maybe it has been done this way on purpose (overridden somewhere by custom code), so by changing the returned value you could break things.
Better to do such things while migrating the module to the next version.